tests/ui/feature-gates/issue-43106-gating-of-builtin-attrs.rs

```diff
@@ -57,7 +57,7 @@
 // see gated-link-args.rs
 // see issue-43106-gating-of-macro_escape.rs for crate-level; but non crate-level is below at "2700"
 // (cannot easily test gating of crate-level #[no_std]; but non crate-level is below at "2600")
-#![proc_macro_derive()] //~ WARN `#[proc_macro_derive]` only has an effect
+#![proc_macro_derive(Test)] //~ WARN `#[proc_macro_derive]` only has an effect
```

Comment:
Note that a breaking change happened here!
Previously a `proc_macro_derive` applied to the crate could have any arguments it wants, it was not checked. This PR fixes this bug, and this now errors.

I discussed this privately with @jdonszelmann, and she said this is fine to change, though we can consider doing a crater run for this just to make sure if you wish.